programs/mpl-core/src/plugins/lifecycle.rs

```diff
@@ -483,6 +507,117 @@ impl Plugin {
+            }
+            Plugin::Edition(edition) => edition.validate_remove_external_plugin(ctx),
+        }
+    }
+
+    /// Route the validation of the update_plugin action to the appropriate plugin.
+    /// There is no check for updating a plugin because the plugin itself MUST validate the change.
+    pub(crate) fn validate_update_external_plugin(
+        plugin: &Plugin,
+        ctx: &PluginValidationContext,
+    ) -> Result<ValidationResult, ProgramError> {
+        let resolved_authorities = ctx
+            .resolved_authorities
+            .ok_or(MplCoreError::InvalidAuthority)?;
+        let base_result = if resolved_authorities.contains(ctx.self_authority) {
+            solana_program::msg!("Base: Approved");
+            ValidationResult::Approved
+        } else {
+            ValidationResult::Pass
+        };
+
+        let result = match plugin {
+            Plugin::Royalties(royalties) => royalties.validate_update_external_plugin(ctx),
+            Plugin::FreezeDelegate(freeze) => freeze.validate_update_external_plugin(ctx),
+            Plugin::BurnDelegate(burn) => burn.validate_update_external_plugin(ctx),
+            Plugin::TransferDelegate(transfer) => transfer.validate_update_external_plugin(ctx),
+            Plugin::UpdateDelegate(update_delegate) => {
+                update_delegate.validate_update_external_plugin(ctx)
+            }
+            Plugin::PermanentFreezeDelegate(permanent_freeze) => {
+                permanent_freeze.validate_update_external_plugin(ctx)
+            }
+            Plugin::Attributes(attributes) => attributes.validate_update_external_plugin(ctx),
+            Plugin::PermanentTransferDelegate(permanent_transfer) => {
+                permanent_transfer.validate_update_external_plugin(ctx)
+            }
+            Plugin::PermanentBurnDelegate(permanent_burn) => {
+                permanent_burn.validate_update_external_plugin(ctx)
+            }
+            Plugin::Edition(edition) => edition.validate_update_external_plugin(ctx),
+        }?;
+
+        match (&base_result, &result) {
+            (ValidationResult::Approved, ValidationResult::Approved) => {
+                Ok(ValidationResult::Approved)
+            }
+            (ValidationResult::Approved, ValidationResult::Rejected) => {
+                Ok(ValidationResult::Rejected)
+            }
+            (ValidationResult::Rejected, ValidationResult::Approved) => {
+                Ok(ValidationResult::Rejected)
+            }
+            (ValidationResult::Rejected, ValidationResult::Rejected) => {
+                Ok(ValidationResult::Rejected)
+            }
+            (ValidationResult::Pass, _) => Ok(result),
+            (ValidationResult::ForceApproved, _) => Ok(ValidationResult::ForceApproved),
+            (_, ValidationResult::Pass) => Ok(base_result),
+            (_, ValidationResult::ForceApproved) => Ok(ValidationResult::ForceApproved),
```

Comment:
I'm fine with leaving this as I know it has parity with internal `validate_update_plugin`, but I don't think `ForceApproved` will be possible with external plugins will it?

Reply:
This is first party plugin validation, though. So it's possible there's a scenario where a first party needs to override a third party.

Reply:
oh yeah I think i got confused by the comment that still said "the plugin itself must validate the change"